app/models/grda_warehouse/tasks/sanity_check_service_history.rb

```diff
@@ -70,6 +69,17 @@ def sanity_check
       end
     end
 
+    def client_still_processing?(client_id)
+      job_ids = Delayed::Job.where(queue: ::ServiceHistory::RebuildEnrollmentsByBatchJob.queue_name, failed_at: nil).
+        jobs_for_class('ServiceHistory::RebuildEnrollments').
+        pluck(:id)
```

Comment:
I feel like this should be a class method in app/jobs/service_history/rebuild_enrollments_job.rb